hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -168,16 +185,19 @@ private void handlePlatformTransaction(
 
             // commit state
             stack.commit();
-        } catch (PreCheckException e) {
+        } catch (final PreCheckException e) {
             recordFailedTransaction(e.responseCode(), recordBuilder, recordListBuilder);
-        } catch (HandleException e) {
+        } catch (final HandleException e) {
             recordFailedTransaction(e.getStatus(), recordBuilder, recordListBuilder);
-        } catch (InterruptedException e) {
-            LOG.error("Interrupted while waiting for signature verification", e);
+        } catch (final InterruptedException e) {
+            logger.error("Interrupted while waiting for signature verification", e);
             Thread.currentThread().interrupt();
             recordBuilder.status(ResponseCodeEnum.UNKNOWN);
-        } catch (Throwable e) {
-            LOG.error("An unexpected exception was thrown during handle", e);
+        } catch (final TimeoutException e) {
+            logger.warn("Timed out while waiting for signature verification, probably going to ISS soon", e);
+            recordBuilder.status(ResponseCodeEnum.UNKNOWN);
+        } catch (final Throwable e) {
```

Comment:
Do we want to handle RuntimeException and Error in different ways (e.g. by exiting on Error)?

Reply:
Didn't we have this conversation already? 🙂 Convince @rbair23, and I will change it.

Reply:
It has come up in several contexts.  I'll always ask the question when I see Throwable, as it's a highly risky thing to catch.  I understand we might need to catch Throwable here; I'm asking more whether we want to handle RuntimeException before this, so we can be more relaxed with RuntimeException and only Errors get caught in this last block.